rest_api/index.adoc

```diff
@@ -7,8 +7,171 @@
 on the OpenShift master servers.
 
-These REST APIs can be used to manage end-user applications, the cluster, and the users of the cluster.
\ No newline at end of file
+These REST APIs can be used to manage end-user applications, the cluster, and the users of the cluster.
+
+== Authentication
+
+API calls must be authenticated with an access token or X.509 certificate. See
+link:../architecture/additional_concepts/authentication.html#api-authentication[authentication architecture]
+for an overview.
+
+This section highlights the token authentication method. With token
+authentication a bearer token must be passed in as an
+link:https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.8[HTTP Authorization header].
+There are two types of access tokens, session and service account.
+
+A **session** token is short-lived, expiring within 24 hours by default. It
+represents a user. The session token may be obtained with this command:
+
+----
+$ oc whoami -t
+dIAo76N-W-GXK3S_w_KsC6DmH3MzP79zq7jbMQvCOUo
+----
+
+**Service account** tokens are long-lived tokens. They are
+link:https://tools.ietf.org/html/rfc7519[JSON Web Token (JWT)] formatted tokens
+and are much longer strings than session tokens. See the
+link:../dev_guide/service_accounts.html#using-a-service-account-s-credentials-externally[Developer Guide]
+for more information.
+
+A service account token may be obtained with these commands:
+
+[options="nowrap"]
+----
+$ oc create serviceaccount robot
+serviceaccount "robot" created
+
+$ oc describe serviceaccount robot
+Name:		robot
+Namespace:	test
+Labels:		<none>
+
+Image pull secrets:	robot-dockercfg-rdrpg
+
+Mountable secrets: 	robot-token-2dsne
+                   	robot-dockercfg-rdrpg
+
+Tokens:            	robot-token-2dsne
+                   	robot-token-9efwm
+
+$  oc describe secret robot-token-2dsne
+Name:		robot-token-2dsne
+Namespace:	test
+Labels:		<none>
+Annotations:	kubernetes.io/service-account.name=robot,kubernetes.io/service-account.uid=ea70e4c7-0663-11e6-b279-fa163e610e01
+
+Type:	kubernetes.io/service-account-token
+
+Data
+====
+token:		fyJhbGciOiJSUzI1NiIyInR5cCI2IkpXVCJ9.EyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJhd2VpdGVrYSIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VjcmV0Lm5hbWUiOiJyb2JvdC10b2tlbi0xcHNuZSIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50Lm5hbWUiOiJyb2JvdCIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50LnVpZCI6ImVhNzBlNGM3LTA2NjMtMTFlNi1iMjc5LWZhMTYzZTYxMGUwMSIsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDphd2VpdGVrYTpyb2JvdCJ9.aGqdv3ajFNguFYwV5eT8Vv-J-fUh8MhFuH82KROqKmQvQcyRU0p7WafrVReAbXXuhArELxtoOdGoMwDWjIiFphEcCE_XNUq2nwhwC5pe2W5yGSthoNb2yOv5cMrK0GWTiFfIYl-pWBigdBOs27cZHcOr5X0tEIVbqXIA_oJYTlzFqnJ6g80SVojfCUC7XZMvxANN9iRZyUSJJ4CW0sXus93UFeWwZCCsm9rtrgwkwklieFwXDm0Py1DkQM6hdS1OouJyROCOgEYcuW4OyR8TKf3WbFD8lMbWqvVG6JLyXCLB-ny3fNlRdl4h434Z6pj-G1zsV42rSr0Nhw45wNLf6g
+ca.crt:		1070 bytes
+namespace:	8 bytes
+----
```

Comment:
Needs an example of how to setup that token in a kubeconfig file:

```
oc login --token="$(oc get secret robot-token-d8trs --template='{{.data.token}}')"
```

However as seen below this doesn't actually work. Even the curl command below doesn't work.